tests/BrowscapTest/Data/Factory/DeviceFactoryTest.php

```diff
@@ -33,12 +33,40 @@ public function testBuildWithoutStandardProperty() : void
         $this->object->build($deviceData, $deviceName);
     }
 
+    /**
+     * @throws \Assert\AssertionFailedException
+     */
+    public function testBuildWithWrongDeviceType() : void
+    {
+        $this->expectException(\UnexpectedValueException::class);
```

Comment:
Just a minor nitpick, but ideally `expectException*` should be just before where the exception happens, rather than at the start of the method. No need to change anything here, but just maybe more logical :)